### PR TITLE
Use default native_constant_invocation rule for PhpCsFixer:risky ruleset

### PR DESCRIPTION
**Pull request by mvorisek**
as in Symfony:risky, fix todo

2nd commit is generated - pure CS fix

### CODE REVIEW

File: src/RuleSet/Sets/PhpCsFixerRiskySet.php [hunk 1]
```diff
@@ -25,18 +25,6 @@ public function getRules()
             '@Symfony:risky' => true,
             'comment_to_phpdoc' => true,
             'final_internal_class' => true,
-            // @TODO: consider switching to `true`, like in @Symfony
```
Comment (keradus) [inline src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: the decision for this action was not yet done.
merging this PR as-is will cause a lot of conflicts between 2.x <> 3.x and lead to conflicts on PRs too

**Review by keradus — CHANGES_REQUESTED** [review]
as commented

File: src/RuleSet/Sets/PhpCsFixerRiskySet.php (hunk 1 above)
Comment (keradus) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: also, default config for this rule changed in v3: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/UPGRADE-v3.md#changed-default-values-of-options

Comment (mvorisek) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: In Symfony:risky ruleset this was changed in 2.x. Thus I propose to change PhpCsFixer:risky ruleset in 2.x as well.

Comment (keradus) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: I understood that intention from PR title,

unfortunately, your comment does not address my concerns

Comment (mvorisek) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: Can I help addressing your concerns? You want this change in 2.x or 3.x only?

Comment (keradus) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: Due to amount of conflicts it gonna create for us, syncing between v2<>v3 and all open PRs, I don't see big value of having this change merged right now.
What would be the benefit overcoming the downsides?

Comment (mvorisek) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: For these reasons, I opened this PR into v2. I belive, FQCN are important as they can be optimized at compile time.

Comment (keradus) [reply src/RuleSet/Sets/PhpCsFixerRiskySet.php:28]: it does not solve the concern for conflicts for all open PRs.

Let's reject the PR for now and re-evaluate this change after we decrease amount of open PRs or stop v2 support

**Comment by mvorisek** [comment]
@keradus why close? The same as we optimize Zend optimized functions to use FQCN, the same we must optimize constants.

**Comment by SpacePossum** [comment]
You can read [here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5469/commits/ef73fa7a0deb73d6af9e19acdd2ad4d98f4ea612#r615860399)  why @keradus closed the PR.

For the project we've to consider the sparse time maintainers have and try to keep away from big conflicts. This is than weighted against the possible upsides of the changes.

>PhpCsFixer ruleset should be not "weaker" than Symfony

The ruleset `PhpCsFixer` really is the internal set for this project, not a SF++ or something. There really are no rules to it. Everyone is free to use it, but it is what it is, so no BC promise, no boards, no docs, no mailing lists, just an internal set for this project :)

>I think constants should be really FQN, like functions.

You can configure it for you own projects, it is not the reason why the PR is merged or not.

**Comment by mvorisek** [comment]
I understand the concerns, but conflicts can be solved easily by applying CS fixer to both branches prior merging easily.

See https://3v4l.org/VcUv2/vld#output and https://3v4l.org/npt1J/vld#output opcodes generated. Using FQ constant names is important for performance as if not used absolutely, constant names are resolved on every access (not even once per file load!) which causes unneeded overhead.

**Pull request closed without merging**